subroutine e
    error "This is an error"
end

subroutine f
    gosub e
end

try
    gosub f
catch e
    print "Caught error: {e.json_encode()}"
end

print "after error"
